addons/account/models/account_move.py

```diff
@@ -1321,36 +1321,122 @@ def _compute_needed_terms(self):
+            account_ids = list(pay_term_lines_by_move.get(move.id, set()))
+            if not account_ids:
+                continue
+                
+            domain = expression.AND([
+                _get_common_domain(move.commercial_partner_id.id),
+                [('account_id', 'in', account_ids)]
+            ])
+            
+            # Bank line domain
+            bank_domain = expression.AND([
+                _get_common_domain(move.commercial_partner_id.id),
+                [
+                    ('account_id.account_type', 'in', ['asset_cash', 'asset_receivable']),
+                    ('journal_id', 'in', bank_journals_by_company.get(move.company_id.id, [])),
+                    ('partner_id', '!=', False),
+                ]
+            ])
+            
             if move.is_inbound():
-                domain.append(('balance', '<', 0.0))
-                payments_widget_vals['title'] = _('Outstanding credits')
+                domain = expression.OR([
+                    expression.AND([domain, [('balance', '<', 0.0)]]),
+                    bank_domain
+                ])
             else:
-                domain.append(('balance', '>', 0.0))
-                payments_widget_vals['title'] = _('Outstanding debits')
-
-            for line in self.env['account.move.line'].search(domain):
+                domain = expression.AND([domain, [('balance', '>', 0.0)]])
+            
+            lines = self.env['account.move.line'].search(domain)
+            all_lines += lines
+            move_line_mapping[move.id] = lines
+        
+        if not all_lines:
+            return
+        
+        move_names = set(all_lines.mapped('move_name'))
+        move_counts = {
+            rec[0]: rec[1]
+            for rec in self.env['account.move.line']._read_group(
+                [('move_name', 'in', list(move_names)), ('id', 'in', all_lines.ids)],
+                ['move_name'], ['__count']
+            )
+        }
+        reconciled_counts = {
+            rec[0]: rec[1]
+            for rec in self.env['account.move.line']._read_group(
+                [('move_name', 'in', list(move_names)), ('reconciled', '=', True)],
+                ['move_name'], ['__count']
+            )
+        }
+        
+        for move in moves_to_process:
+            lines = move_line_mapping.get(move.id, self.env['account.move.line'])
+            if not lines:
+                continue
+                
+            payments_widget_vals = {
+                'outstanding': True,
+                'content': [],
+                'move_id': move.id,
+                'title': _('Outstanding credits') if move.is_inbound() else _('Outstanding debits')
+            }
+            
+            for line in lines:
+                if (
+                    line.account_id.account_type == 'asset_cash' and
+                    (move_counts.get(line.move_name, 0) > 1 or reconciled_counts.get(line.move_name, 0))
```

Comment:
Could you explain what lines we're preventing here?
I must admit I don't understand fully what the goal of this is.